x-pack/plugins/canvas/canvas_plugin_src/renderers/time_filter/index.js

```diff
@@ -16,22 +16,23 @@ export const timeFilter = () => ({
   help: 'Set a time window',
   reuseDomNode: true, // must be true, otherwise filters get reset when re-rendered
   render(domNode, config, handlers) {
-    const ast = fromExpression(handlers.getFilter());
+    const filterExpression = handlers.getFilter();
+    const filterExists = filterExpression !== '';
+    let filter = '';
+    if (filterExists) {
+      const ast = fromExpression(filterExpression);
+      filter = toExpression(ast);
```

Comment:
This has to go after the AST update down on line 31, otherwise it could end up being the old value. 

Alternatively, you I think you can leave `toExpression(ast)` in the component prop, since an empty expression's AST is `[]` and `toExpression([])` *should* produce an empty string (which is what the incorrect filter value would still be).

Reply:
good point, no need for conditionalization there!